lib/cortex/faraday_middleware/encode_uri_path.rb

```diff
@@ -0,0 +1,14 @@
+require 'faraday/middleware'
+require 'addressable/uri'
+
+module Cortex
+  module FaradayMiddleware
+    class EncodeURIPath < Faraday::Middleware
+      def call(env)
+        env[:url].path = Addressable::URI.encode(env[:url].path)
```

Comment:
Blindly encoding sets you up for double encoding -- example:

```rb
[10] pry(main)> x = Addressable::URI.parse('http://www.careerbuilder.com/as df?sadfsa=tes$t')
=> #<Addressable::URI:0x3fbfd66941d4 URI:http://www.careerbuilder.com/as df?sadfsa=tes$t>
[11] pry(main)> x.path
=> "/as df"
[12] pry(main)> Addressable::URI.encode(x.path)
=> "/as%20df"
[13] pry(main)> Addressable::URI.encode(Addressable::URI.encode(x.path))
=> "/as%2520df"
```

I think noramlize_component might be better? https://github.com/sporkmonger/addressable/blob/master/lib/addressable/uri.rb#L461